### PR TITLE
fix: making mint button active when not connected

### LINKED ISSUE #6185: making mint button active when not connected
ref #6163 

sry @vikiival :D 

- should work the same as the buy button in the gallery item - being in the active state even when the wallet is not connected - after being touched, it should open the wallet connect dialog. 

- situations when the button is inactive:
   - minted out drop
   - reached nft limit

image

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Needs Design check

- @exezbcz please review

## Context

- [x] Closes #6185
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

## Screenshot 📸

- [ ] My fix has changed UI

### <samp>🤖 Generated by Copilot at 2ebc940</samp>

Added a modal to connect wallet before minting unlockable NFTs. Refactored `NeoButton` component to support modal functionality.

### <samp>🤖 Generated by Copilot at 2ebc940</samp>

> _To mint an NFT with a lock_
> _You need to connect to your block_
> _So a modal was made_
> _With a `NeoButton` to aid_
> _The user in choosing their stock_

### CODE REVIEW

File: components/collection/unlockable/UnlockableContainer.vue [hunk 1]
```diff
@@ -170,11 +171,16 @@ import { timeAgo } from '@/components/collection/utils/timeAgo'
...
 import { endOfHour, startOfHour } from 'date-fns'
+import type Vue from 'vue'
+import { ConnectWalletModalConfig } from '@/components/common/ConnectWallet/useConnectWallet'
+
 const { toast } = useToast()
 
 const Loader = defineAsyncComponent(
   () => import('@/components/shared/Loader.vue')
 )
+const { $buefy } = useNuxtApp()
```
Comment (roiLeo) [inline components/collection/unlockable/UnlockableContainer.vue:182]: I would like to see something else, please consider refactoring ref #5852

Comment (Jarsen136) [reply components/collection/unlockable/UnlockableContainer.vue:182]: I guess buefy modal component should be refactored on issue #5852 . For now, we have to use it to open the wallet modal.

**Comment by daiagi** [comment]
@exezbcz 
1. should the Mint button be disabled when not in the minting window?
image

2. @Jarsen136 @vikiival 
there is no notification that minting didn't happen
3. @Jarsen136 @vikiival 
out of the scope of this PR i guess, but if the logged in account can't mint would it be better/possible to disable the mint button with a tooltip instead of making user wait through the loader for nothing?

**Comment by Jarsen136** [comment]
> @exezbcz
> 
> 1. should the Mint button be disabled when not in the minting window?
>    image

It means the end time of the minting window actually. I will change the misleading text as mentioned at the issue #6186

> 2. @Jarsen136 @vikiival
>    there is no notification that minting didn't happen

There is a notification when the mint failed. I have tried to mint some times, and it works though there is no notification about success information. I have added the notification when minting is successful.

> 3. @Jarsen136 @vikiival
>    out of the scope of this PR i guess, but if the logged in account can't mint would it be better/possible to disable the mint button with a tooltip instead of making user wait through the loader for nothing?
> 
> so generally i think it would be better UX if the button is disabled: a. outside of minting window b. when the connected wallet is not allowed to mint
> 
> wdyt?

It's a good idea to show the reason why the button is disabled. But, the tooltip may not be friendly on mobile I guess.

**Comment by exezbcz** [comment]
> 1. should the Mint button be disabled when not in the minting window?

yes, it should, but since we have 60-minute windows, it's not necessary.



> 3\. out of the scope of this PR i guess, but if the logged in account can't mint would it be better/possible to disable the mint button with a tooltip instead of making user wait through the loader for nothing?

Best case would be to have a tooltip for both cases. 
- when the user Is not allowed to mint, it will show that only one per user
- if the minting window ends, it shows either add to calendar or the disabled button saying minting window is not active
- since most of the user will use this page on mobile, tooltips are useless
- 

> so generally i think it would be better UX if the button is disabled: a. outside of minting window b. when the connected wallet is not allowed to mint

yes

**Review by vikiival — COMMENTED** [review]
Looks oki

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $20 @ 4.29 USD/DOT ~ 4.662 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0x1f3031201240d51f04d25ec1678a1e84a36e246c50c391abff9235e0c2750b7f](https://polkadot.subscan.io/extrinsic/0x1f3031201240d51f04d25ec1678a1e84a36e246c50c391abff9235e0c2750b7f)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**